spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/support/S3SignedRedirectRequestController.java

```diff
@@ -68,7 +69,7 @@ public ResponseEntity<Map<String, String>> getBlobRedirect(@RequestHeader("Autho
 
 	@RequestMapping("/test/docker/registry/v2/blobs/test/data")
 	public ResponseEntity<Resource> getSignedBlob(@RequestHeader Map<String, String> headers) {
-		if (headers.containsKey("authorization")) {
+		if (!headers.containsKey("authorization")) {
```

Comment:
I'm curious why it worked in 2.11.x  when the test said it should not contain `authorized` but now requires  authorized?

Reply:
The question is then how does the authorization header not get dropped?

Reply:
`org.springframework.cloud.dataflow.container.registry.authorization.DropAuthorizationHeaderRequestRedirectStrategy` doesn't reach the code to check if a header should be dropped.

Reply:
Comparing [org.apache.hc.client5.http.protocol.RedirectStrategyorg.apache.http.client.RedirectStrategy](https://hc.apache.org/httpcomponents-client-5.4.x/current/httpclient5/apidocs/org/apache/hc/client5/http/protocol/RedirectStrategy.html)
to [org.apache.http.client.RedirectStrategy](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/RedirectStrategy.html)
shows the return type change from `HttpUriRequest` to `URI`
So we will have to figure out where we can actually deal with the headers and if we need to actually drop the `Authorization` header.

Reply:
The initial change I made was invalid and reversed.
The `Authorization` header should be dropped for most of the redirects on container queries.